common/utils.py

```diff
@@ -217,34 +217,37 @@ def get_contract_instance(base_path, contract_address, contract_name):
     return web3_object.eth.contract(abi=contract_abi, address=contract_address)
 
 
-def verify_signature(airdrop_id, airdrop_window_id, address, signature):
+def verify_signature(airdrop_id, airdrop_window_id, address, signature, block_number):
     public_key = recover_address(
-        airdrop_id, airdrop_window_id, address, signature)
+        airdrop_id, airdrop_window_id, address, signature, block_number)
 
     if public_key.lower() != address.lower():
-        logger.info(f"Invalid signature")
+        logger.info(f"INVALID SIGNATURE {signature}")
+        logger.info(f"For airdrop_id:{airdrop_id} , airdrop_window_id:{airdrop_window_id} , address:{address} , "
+                    f"block_number{block_number}")
         #raise Exception("Invalid signature")
 
 
-def recover_address(airdrop_id, airdrop_window_id, address, signature):
+def recover_address(airdrop_id, airdrop_window_id, address, signature, block_number):
     address = Web3.toChecksumAddress(address)
     message = web3.Web3.soliditySha3(
-        ["uint256", "uint256", "address"],
-        [int(airdrop_id), int(airdrop_window_id), address],
+        ["uint256", "uint256", "uint256", "address"],
```

Comment:
@rajeevbbqq , please confirm this part

Reply:
@anandrgitnirman dApp is now creating signature with sign typed data.